[Inductor] ExternKernelBenchmarkRequest best attempt

PaulZhang12 opened the pull request:
Differential Revision: D88759335

Fixes some issues caused by https://github.com/pytorch/pytorch/pull/169605. Best effort at constructing input/output tensor meta for ExternKernel bmreq


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

eellison reviewed (approved):
let's try to fix in follow up

PaulZhang12 committed:
[Inductor] ExternKernelBenchmarkRequest best attempt (#169971)

Summary:


Sometimes input tensor meta/output tensor meta fails construction. In this case, we are ok with continuing on with compilation. If input tensor meta/output tensor meta is required for benchmarking, then we fail.

Test Plan: Fixes S600787

Reviewed By: eellison, MackZackA

Differential Revision: D88759335
